http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -1049,27 +1038,21 @@ public void close()
             extends Listener.Adapter
     {
         private final JettyResponseFuture<?, ?> future;
-        private final int maxLength;
 
         @GuardedBy("this")
         private byte[] buffer = new byte[0];
```

Comment:
Instead of using a single `byte[]` use a `List<Slice>` and use a modified version of `ChunkedSliceInput`.  Also the I would grow the buffered chunk from 1k to 1MB over time.

Reply:
In my test case, the "Remote page is too large" issue sometimes can be resolved by 

`set session exchange_compression = true;`

Hope this makes sense too.